CardForm/src/test/java/com/braintreepayments/cardform/utils/CardNumberTransformationTest.java

```diff
@@ -0,0 +1,32 @@
+package com.braintreepayments.cardform.utils;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+
+import java.util.Arrays;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(RobolectricTestRunner.class)
+public class CardNumberTransformationTest {
+
+    CardNumberTransformation mTransform = new CardNumberTransformation();
+
+    @Test
+    public void masksCardNumber() {
+        String transformed = (String) mTransform.getTransformation("4111111111111111", null);
+
+        char[] padding = new char[7];
+        Arrays.fill(padding, Character.MIN_VALUE);
+        String expected = new String(padding) + "•••• 1111";
+        assertEquals(expected, transformed);
```

Comment:
I think I prefer just the hardcoded expectation rather than generated. Helps keep the story simple on what this is supposed to look like after a transform.

Reply:
It's a _little_ weird since we have these zero-space chars in there also. It would basically look like `assertEquals("\u0000\u0000\u0000\u0000\u0000\u0000\u0000•••• 1111", transformed);` but I think this is fine.